conans/client/generators/__init__.py

```diff
@@ -31,10 +31,11 @@ def _get_generator_class(generator_name):
     try:
         generator_class = _generators[generator_name]
         # This is identical to import ... form ... in terms of cacheing
-        return getattr(importlib.import_module(generator_class), generator_name)
     except KeyError as e:
         raise ConanException(f"Invalid generator '{generator_name}'. "
                              f"Available types: {', '.join(_generators)}") from e
+    try:
+        return getattr(importlib.import_module(generator_class), generator_name)
```

Comment:
👍

Reply:
yes, my IDE linter was warning against possible undefinition of variables, which might happen when a multi-line ``try`` clause can raise different exceptions in different lines. I try to remove these linter warnings when possible, as I have learned that often they can be a bug waiting for a corner case to become an issue.